osrsbox/items_api/item_definition.py

```diff
@@ -0,0 +1,318 @@
+"""
+Author:  PH01L
+Email:   phoil@osrsbox.com
+Website: https://www.osrsbox.com
+
+Copyright (c) 2019, PH01L
+
+###############################################################################
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+"""
+
+import os
+import json
+import datetime
+import collections
+from typing import List
+from typing import Dict
+
+import safe_cast
+
+from osrsbox.items_api import item_stats
+from osrsbox.items_api import item_equipment
+
+
+class ItemDefinition:
+    """This class defines the object structure and properties for an OSRS item.
+
+    The ItemDefinition class is the object that retains all properties and stats
+    for one specific item. Every item has the properties defined in this class.
+    Equipable items have additional properties defined in the linked ItemStats
+    and ItemEquipment classes.
+    """
+    def __init__(self):
+        self.properties = [
+            "id",
+            "name",
+            "members",
+            "tradeable",
+            "tradeable_on_ge",
+            "stackable",
+            "noted",
+            "noteable",
+            "linked_id",
+            "equipable",
+            "cost",
+            "lowalch",
+            "highalch",
+            "weight",
+            "buy_limit",
+            "quest_item",
+            "release_date",
+            "examine",
+            "url"]
+
+        # Initialize an empty ItemStats object
+        self.item_stats: object = item_stats.ItemStats()
+        # Initialize an empty ItemEquipment object
+        self.item_equipment: object = item_equipment.ItemEquipment()
+        # Initialize an empty output dictionary for an item
+        self.json_out: Dict = collections.OrderedDict()
+
+    @property
+    def id(self) -> int:
+        """The ID number of the item."""
+        return self._id
+
+    @id.setter
+    def id(self, value: int):
+        self._id = safe_cast.safe_cast(value, int, default=None)

```

Comment:
You are casting it to an `int` and declaring that the `type` of `value` is `int`. The types should be what the code is going to pass in / what is expected of the user to pass in. I think `value` can be a string as well otherwise we wouldn't have to do the int conversion? Not sure of the specifics here.